### ISSUE TITLE
BUG in TST: towncrier fails on dependent PRs

### ISSUE THREAD

**Issue by nstarman** [astropy/astropy#11578]
### Description

In #11444 I'm getting 2 towncrier checks and one's failing.

<img width="862" alt="Screen Shot 2021-04-16 at 00 30 11" src="https://user-images.githubusercontent.com/8949649/114971739-021b2000-9e4b-11eb-8e29-4db2325e96a7.png">

### Steps to Reproduce

Make a branch. Open a PR with a changelog. Make a branch from the first branch. Open a PR with a changelog.

@saimn @pllim

**Comment by nstarman** [comment]
I've broken around the issue by deleting the changelog for the old PR in the new PR. This is not good, because I'll have to remember to undelete it.

**Comment by pllim** [comment]
@nstarman , did rebase get rid of the duplicate check? There are two because I changed the bot rule so it is unskippable. This is a temporary unfortunate side effect. There is a chance it won't go away either.

Your case is so special that I am not sure what we can do about it. You can add a check box on your original post of the PR to remind yourself to take care of the change log. I apologize for any inconvenience caused!

**Comment by pllim** [comment]
Also when I check #11444 now, I only see one check and it is green. Is the green status now good or bad? 🤷 

`Changelog file (docs/changes/coordinates/11444.feature.rst) correctly added for PR 11444.`

**Comment by nstarman** [comment]
Rebasing onto main would get ride of the check. The problem seems to be that if one opens a PR dependent on another branch that has an active PR then both PRs will fail towncrier. I "fixed" the issue by deleting the changelog from my new PR, but the problem is that if I don't remember to restore it and the PR gets merged, it will delete the old PR's changelog. This may be an uncommon problem, but it will be encountered by anyone working on followups to an active PR.

**Comment by pllim** [comment]
I am not sure if there is a simple fix for this on the bot side without overcomplicating the check logic for a use case that would rarely happen. Suggestions welcome.

**Comment by nstarman** [comment]
Hmm. It's a balance. One the one hand, the fix would be that if a PR had a changelog for itself, it wouldn't fail if it also had a changelog for another PR. On the other hand, that's a good check to have to find mistakes in non-dependent PRs.

A possible resolution is if towncrier checked for a "Depends on #____" in the opening comment. Were that present, it wouldn't fail if it also had a changelog for another PR. Else, it would.

**Comment by nstarman** [comment]
What's the link to the towncrier bot repo?

**Comment by pllim** [comment]
The canonical check is at https://github.com/OpenAstronomy/baldrick/blob/master/baldrick/plugins/github_towncrier_changelog.py . But that currently is not running (#11481), so what you see is the stop-gap solution at https://github.com/pllim/actions-towncrier-changelog/blob/main/check_changelog.py .

I'd say you can take a stab at `baldrick/plugins/github_towncrier_changelog.py` and whatever that is accepted there can be easily ported to the stop-gap solution.

But let's wait to hear back from @Cadair first.

**Comment by nstarman** [comment]
@pllim. Has the situation changed with baldrick and the stop-gap solution? I was thinking of taking a crack at this because I have a fair number of dependent PRs.

**Comment by pllim** [comment]
> Has the situation changed

No. Still using the Action one.

**Comment by pllim** [comment]
Haven't seen this in a while.
